libs/librepcb/projecteditor/boardeditor/fsm/bes_select.cpp

```diff
@@ -759,6 +771,91 @@ bool BES_Select::removeSelectedItems() noexcept {
   }
 }
 
+void BES_Select::measureLengthInDirection(bool              directionBackwards,
+                                          const BI_NetLine& netline,
+                                          QSet<Uuid>&       visitedSegments,
+                                          UnsignedLength&   totalLength) {
+  auto currentNetline = std::cref(netline);
+  auto currentAnchor  = std::cref(directionBackwards ? netline.getStartPoint()
+                                                    : netline.getEndPoint());
+
+  for (;;) {
+    tl::optional<BI_NetLine&> nextNetline = tl::nullopt;
+    for (BI_NetLine* nl : currentAnchor.get().getNetLines()) {
+      // Don't visit a segment twice
+      if (visitedSegments.contains(nl->getUuid())) {
+        continue;
+      }
+      // Only visit selected segments
+      if (nl->isSelected()) {
+        if (nextNetline.has_value()) {
+          // There's already another connected and selected netline
+          throw LogicError(__FILE__, __LINE__,
+                           tr("Selected segment may not have branches"));
```

Comment:
Add a period (or exclamation mark) at the end of sentences (it's shown in a message box and looks strange without the period):

```suggestion
                           tr("Selected segment may not have branches."));
```